apps/s_client.c

```diff
@@ -1432,13 +1428,20 @@ int s_client_main(int argc, char **argv)
         goto end;
     }
 
-    if (split_send_fragment > SSL3_RT_MAX_PLAIN_LENGTH) {
-        BIO_printf(bio_err, "Bad split send fragment size\n");
+    if ((max_send_fragment > 0 && max_send_fragment < 512)
+         || max_send_fragment > SSL3_RT_MAX_PLAIN_LENGTH ) {
+        BIO_printf(bio_err, "%s: Invalid max send fragment size\n", prog);
+        goto end;
+    }
+
+    if ((split_send_fragment > 0 && split_send_fragment < 512)
+         || split_send_fragment > SSL3_RT_MAX_PLAIN_LENGTH ) {
+        BIO_printf(bio_err, "%s: Invalid split send fragment size\n", prog);
         goto end;
     }
 
     if (max_pipelines > SSL_MAX_PIPELINES) {
-        BIO_printf(bio_err, "Bad max pipelines value\n");
+        BIO_printf(bio_err, "%s: too large max-pipelines value\n", prog);
         goto end;
     }
```

Comment:
Ok, all these checks are a bit over the top.  What happens when the libssl code changes those ranges, will we have to keep track of that and update the values here as well?  It should be enough that `SSL_CTX_set_max_send_fragment()`, `SSL_CTX_set_split_send_fragment()`, `SSL_CTX_set_max_pipelines()` and SSL_ friends return 0 when the given argument is out of range...